src/game/behaviors/spawn_star.inc.c

```diff
@@ -164,8 +164,15 @@ void bhv_hidden_red_coin_star_init(void) {
         o->activeFlags = ACTIVE_FLAG_DEACTIVATED;
     }
 
-    o->oHiddenStarTriggerTotal = numRedCoinsRemaining + gRedCoinsCollected;
-    o->oHiddenStarTriggerCounter = o->oHiddenStarTriggerTotal - numRedCoinsRemaining;
+    if (o->oBehParams2ndByte != 0) {
+        o->oHiddenStarTriggerTotal = o->oBehParams2ndByte;
+        o->oHiddenStarTriggerCounter = gRedCoinsCollected;
+    }
+    else {
+        o->oHiddenStarTriggerTotal = numRedCoinsRemaining + gRedCoinsCollected;
+        o->oHiddenStarTriggerCounter = o->oHiddenStarTriggerTotal - numRedCoinsRemaining;
+    }
+    
 }
```

Comment:
This function here doesn't work because of the `numRedCoinsRemaining == 0` check (out of range from what I can select). The conditions for spawning in a star on init should also change depending on bparam2.

Reply:
oh duh cant believe i missed that